### ISSUE TITLE
Assertion `plist->present & PP_PARTICIPANT_LEASE_DURATION' can be remotely triggered

### ISSUE THREAD

**Issue by squizz617** [eclipse-cyclonedds/cyclonedds#1758]
Hi, reporting another remotely reachable assertion failure.
Similar to #1751, this can be triggered after a publisher and a subscriber complete exchanging their participant discovery packets.

* Triggered assertion ([code](https://github.com/eclipse-cyclonedds/cyclonedds/blob/0.10.2/src/core/ddsc/src/dds_serdata_builtintopic.c#L166)):
```
  assert (plist->present & PP_PARTICIPANT_LEASE_DURATION);
```

* DATA submessage triggering the assertion failure:
```
0000   15 05 00 00 00 00 10 00 00 01 00 c7 00 01 00 c2
0010   00 00 00 00 01 00 00 00 00 03 00 00 15 00 04 00
0020   02 02 00 00 16 00 04 00 01 0f 00 00 50 00 10 00
0030   01 0f 11 3e f6 42 cd 90 00 00 00 00 00 00 01 c1
0040   32 00 18 00 01 00 00 00 f2 1c 00 00 00 00 00 00
0050   00 00 00 00 00 00 00 00 0a 00 00 0f 31 00 18 00
0060   01 00 00 00 f3 1c 00 00 00 00 00 00 00 00 00 00
0070   00 00 00 00 0a 00 00 0f 82 00 08 00 14 00 00 00
0080   00 00 00 00 58 00 04 00 3f 0c 00 00 62 00 10 00
0090   0a 00 00 00 70 75 62 6c 69 73 68 65 72 00 00 00
00a0   59 00 28 00 01 00 00 00 11 00 00 00 50 41 52 54
00b0   49 43 49 50 41 4e 54 5f 54 59 50 45 00 00 00 00
00c0   07 00 00 00 53 49 4d 50 4c 45 00 00 01 00 00 00
```

* Wireshark dissection of the submessage:
```
submessageId: DATA (0x15)
    Flags: 0x05, Data present, Endianness bit
    octetsToNextHeader: 0
    0000 0000 0000 0000 = Extra flags: 0x0000
    Octets to inline QoS: 16
    readerEntityId: ENTITYID_BUILTIN_PARTICIPANT_READER (0x000100c7)
    writerEntityId: ENTITYID_BUILTIN_PARTICIPANT_WRITER (0x000100c2)
    writerSeqNumber: 1
    serializedData
        encapsulation kind: PL_CDR_LE (0x0003)
        encapsulation options: 0x0000
        serializedData:
            PID_PROTOCOL_VERSION
                parameterId: PID_PROTOCOL_VERSION (0x0015)
                parameterLength: 4
                Protocol version: 2.2
            PID_VENDOR_ID
                parameterId: PID_VENDOR_ID (0x0016)
                parameterLength: 4
                vendorId: 01.15 (eProsima - Fast-RTPS)
            PID_PARTICIPANT_GUID
                parameterId: PID_PARTICIPANT_GUID (0x0050)
                parameterLength: 16
                Participant GUID: 010f113e f642cd90 00000000 000001c1
            PID_METATRAFFIC_UNICAST_LOCATOR (LOCATOR_KIND_UDPV4, 10.0.0.15:7410)
                parameterId: PID_METATRAFFIC_UNICAST_LOCATOR (0x0032)
                parameterLength: 24
                locator
            PID_DEFAULT_UNICAST_LOCATOR (LOCATOR_KIND_UDPV4, 10.0.0.15:7411)
                parameterId: PID_DEFAULT_UNICAST_LOCATOR (0x0031)
                parameterLength: 24
                locator
            Unknown (0x0082)
                parameterId: Unknown (0x0082)
                parameterLength: 8
                parameterData: 1400000000000000
            PID_BUILTIN_ENDPOINT_SET
                parameterId: PID_BUILTIN_ENDPOINT_SET (0x0058)
                parameterLength: 4
                Flags: 0x00000c3f, Participant Message DataReader, Participant Message DataWriter, Subscription Detector, Subscription Announcer, Publication Detector, Publication Announcer, Participant Detector, Participant Announcer
            PID_ENTITY_NAME
                parameterId: PID_ENTITY_NAME (0x0062)
                parameterLength: 16
                entityName: publisher
            PID_PROPERTY_LIST (1 properties)
                parameterId: PID_PROPERTY_LIST (0x0059)
                parameterLength: 40
                Property List
                    Property Name: PARTICIPANT_TYPE
            PID_SENTINEL
                parameterId: PID_SENTINEL (0x0001)
```

* Stderr:
```
HelloworldSubscriber: /home/seulbae/ddssecurity/targets/cyclonedds-0.10.2/src/core/ddsc/src/dds_serdata_builtintopic.c:166: void translate_pp_lease_duration(dds_qos_t *, const ddsi_plist_t *): Assertion `plist->present & PP_PARTICIPANT_LEASE_DURATION' failed.
Aborted
```

* Packet capture from the incident:
[cyclonedds-assert-20230706.pcap.zip](https://github.com/eclipse-cyclonedds/cyclonedds/files/11971210/cyclonedds-assert-20230706.pcap.zip)

I haven't figured the exact cause for this yet, but reporting as you'll be able to quickly do that. Please let me know if any additional information is needed. Thank you.

**Comment by eboasson** [comment]
If you could have a look at #1759, that'd be great!

**Comment by squizz617** [comment]
Confirmed that #1759 fixes this and #1751! Thank you.

**Comment by eboasson** [comment]
Thanks for confirming @squizz617 ☺️
